src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1589,27 +1603,37 @@ void Partitioner::optimize(const std::string& func_name) {
             }
             ov::parallel_for(func_group.refs.size(), [&](std::size_t f_idx) {
                 auto& funcall = func_group.refs[f_idx].get();
-                std::vector<ov::Tensor> to_concat;
+                std::vector<LazyTensor> to_concat;
+                // Fill tensor vector
                 for (auto&& cidx : to_concat_idx) {
-                    to_concat.push_back(funcall._closure[cidx]);
+                    to_concat.push_back(funcall._transformations[cidx]);
+                }
+                // Set to lazy tensor history
+                for (auto&& cidx : to_concat_idx) {
+                    // FIXME: Assuming here concat goes first and other transformations later.
+                    //        This allows to store ov::Tensor and ignore their potential history of transformations
+                    funcall._transformations[cidx].update(TransformType::CONCAT, std::make_pair(to_concat, axis));
+                }
+                // Pick the first (could be any) LazyTensor and set as new future-concatenated tensor
+                if (!to_concat.empty()) {
+                    funcall._transformations.push_back(to_concat.front());
```

Comment:
Oh I see why, but let's don' do it like this

Reply:
Done